social/strategies/django_strategy.py

```diff
@@ -63,7 +62,14 @@ def request_path(self):
 
     def request_port(self):
         """Port in use for this request"""
-        return self.request.META['SERVER_PORT']
+        try:  # django >= 1.9
+            return self.request.get_port()
+        except AttributeError:  # django < 1.9
+            host_parts = self.request.get_host().split(':')
+            try:
+                return host_parts[1]
+            except IndexError:
+                return None
```

Comment:
So shouldn't the `except AttributeError:` path just return `self.request.META['SERVER_PORT']` then instead of splitting the host?

Or does looking at the port from get_host() emulate using `USE_X_FORWARDED_PORT`?  In that case, it seems like the `except IndexError:` should still fall back to `self.request.META['SERVER_PORT']` so we get at least `80` or `443` rather than `None`?